translate/filters/test_checks.py

```diff
@@ -1374,3 +1374,124 @@ def test_skip_checks_per_language_in_some_checkers():
     failures = stdchecker.run_filters(unit)
     # But it is not in StandardChecker.
     assert 'accelerators' in failures.keys()
+
+
+def test_mozilla_no_accelerators():
+    """Test no-accelerators check in MozillaChecker."""
+    mozillachecker = checks.MozillaChecker()
+    assert fails(mozillachecker.noaccelerators, "&File", "&Fayile")
+    assert fails(mozillachecker.noaccelerators,
+                 "Warn me if this will disable any of my add&-ons",
+                 "&Waarsku my as dit enige van my byvoegings sal deaktiveer")
+    assert passes(mozillachecker.noaccelerators, "&File", "Fayile")
+    assert fails(mozillachecker.noaccelerators, "File", "&Fayile")
+    assert passes(mozillachecker.noaccelerators, "Mail &amp; News",
+                  "Pos en Nuus")
+    assert fails(mozillachecker.noaccelerators, "Mail &amp; News",
+                 "Pos en &Nuus")
+    assert passes(mozillachecker.noaccelerators, "&File", "Fayile")
+
+
+def test_noaccelerators_only_in_mozilla_checker():
+    """Test no-accelerators check for some languages in Mozilla checker."""
+    from translate.storage import base
+
+    asmozillachecker = checks.MozillaChecker(
+        checkerconfig=checks.CheckerConfig(targetlanguage="as")
+    )
+    glmozillachecker = checks.MozillaChecker(
+        checkerconfig=checks.CheckerConfig(targetlanguage="gl")
+    )
+    stdchecker = checks.StandardChecker(
+        checkerconfig=checks.CheckerConfig(accelmarkers="&", targetlanguage="as")
+    )
+
+    # No-accelerators check passes.
+    # No-accelerators check not in StandardChecker so it passes.
+    str1, str2, __ = strprep(u"&Check for updates", u"আপডেইটসমূহৰ বাবে নিৰীক্ষণ কৰক")
+    unit = base.TranslationUnit(str1)
+    unit.target = str2
+
+    failures = asmozillachecker.run_filters(unit)
+    assert 'noaccelerators' not in failures.keys()
+    failures = glmozillachecker.run_filters(unit)
```

Comment:
No, because here we are testing that `noaccelerators` in only in `MozillaChecker`, and only for some languages. So we test for:

- `MozillaChecker` with language that uses `noaccelerators`
- `MozillaChecker` with language that does not use `noaccelerators`
- `StandardChecker` which never uses `noaccelerators`